scripts/fedora/openstack.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash -eux
+
+if [ $(uname -m)=="ppc64" -o $(uname -m)=="ppc64le" ] ; then
+    dnf -y install ppc64-diag
+fi
+
+if [ -e /usr/bin/dnf ] ; then
+  dnf -y install cloud-init cloud-utils yum
+else
+  yum -y install cloud-init cloud-utils dracut-modules-growroot
+fi
+dracut -f
+
+if [ -e /boot/grub/grub.conf ] ; then
+  sed -i -e 's/rhgb.*/console=ttyS0,115200n8 console=tty0 quiet/' /boot/grub/grub.conf
+  sed -i -e 's/^timeout=.*/timeout=0/' /boot/grub/grub.conf
+  cd /boot
+  ln -s boot .
+elif [ -e /etc/default/grub ] ; then
+  # output bootup to serial
+  sed -i -e \
+    's/GRUB_CMDLINE_LINUX=\"\(.*\)/GRUB_CMDLINE_LINUX=\"console=ttyS0,115200n8 console=tty0 \1/g' \
+    /etc/default/grub
+  # No timeout for grub menu
+  sed -i -e 's/^GRUB_TIMEOUT.*/GRUB_TIMEOUT=0/' /etc/default/grub
+  # No fancy boot screen
+  grep -q rhgb /etc/default/grub && sed -e 's/rhgb //' /etc/default/grub
```

Comment:
Is the grep necessary here? if `sed` runs when there is no match, then the file shouldn't be modified

Reply:
Actually it does the opposite :)

Reply:
Now I'm very confused, what is this line actually doing?

AFAICT it's testing if `rhgb` exists in `/etc/default/grub`, and if it does, it removes `rhgb` and a space, but `sed` doesn't run in place, so it's just spitting the result to STDOUT?

Reply:
Hmm, good point, it does need to have ``-i`` added.